lib/MooseX/Types/Combine.pm

```diff
@@ -108,6 +152,13 @@ sub _provided_types {
     %$types;
 }
 
+sub type_names {
```

Comment:
This is a new public method without documentation. Should it be private or documented? I'm inclined to say it should be documented, since it seems like a nice addition.